### ISSUE TITLE
Unify the terminology of "User Profiles" vs "Attributes"

### ISSUE THREAD

**Issue by chpapa** [authgear/authgear-server#4100]
Notice that right now we use "User Profiles" and "Attributes" changeably. (e.g. look at our doc)

<img src="https://uploads.linear.app/860e83ae-05f0-4c4b-bd93-8a4b8d251836/8d384262-1830-4647-8e5a-f08838dd8439/bb2604de-d0e3-43d8-a6a6-f31b3a6cd10a?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzg2MGU4M2FlLTA1ZjAtNGM0Yi1iZDkzLThhNGI4ZDI1MTgzNi84ZDM4NDI2Mi0xODMwLTQ2NDctOGU1YS1mMDg4MzhkZDg0MzkvYmIyNjA0ZGUtZDBlMy00M2Q4LWE2YTYtZjMxYjNhNmNkMTBhIiwiaWF0IjoxNzEyNTQ3MjIwLCJleHAiOjMzMjgzMTA3MjIwfQ.TCOFldBpTqzeTn3K6h0vB_iVfdlpc24n3DJxyNmkb0c " alt="SCR-20240408-g1u.png" width="600" height="350" />

I believe the confusion comes from:

* On Admin Portal or when explain to end users, it is easier to use the term "User Profiles"
* However, in Admin API or user data, we called it `standardAttributes` or `customAttributes`

Assigned it to @fungc-io first, as I believe this issue need some discussion before deciding the next step.

I think it would be better if we:

* Always call it User Profiles, in documentation and in our APIs
* Except when the use case is OIDC, and **\*\*IF OIDC DOES USE THE TERM ATTRIBUTES\*\*,** than we use attributes whenever it is OIDC specific.

Thoughts? @louischan @fungc-io 

cc: @pius

**Comment by louischan-oursky** [comment]
In OIDC, they are called Standard Claims.

**Comment by chpapa** [comment]
So maybe we should drop the term attributes at all?

**Comment by louischan-oursky** [comment]
But we have been using the term attributes in public API like in [configuration](https://github.com/authgear/authgear-server/blob/main/docs/specs/user-profile/design.md), [GraphQL schema](https://github.com/authgear/authgear-server/blob/main/portal/src/graphql/adminapi/schema.graphql#L1539), and [user import API](https://github.com/authgear/authgear-server/blob/main/docs/specs/user-import.md#update-behavior-of-each-field), etc. It is hard to drop it now.

In the [spec](https://github.com/authgear/authgear-server/blob/main/docs/specs/user-profile/design.md#user-profile), User profile and attributes are defined as

> User profile consists of standard attributes, custom attributes and roles.

So `user profile = standard attributes + custom attributes`. I think we can stick with the term "user profile" if we refer to any attributes, and use "custom attributes" when we talk about the use cases that make use of custom attributes.

**Comment by tung2744** [comment]
I guess what we are actually confused is `standard attributes` vs `standard claims`?

**Comment by chpapa** [comment]
attributes or profiles vs claims would be a bit confusing. But I think this is unavoidable as claims is used by OIDC.

User Profiles and Attributes is the confusion that we introduce on our own.

BTW, I think we should really avoid anything that's confusing like "User Profile = standard attributes + custom attributes" in future, that's just creating confusion to everyone.

I think if we procasinate it, we should at least make sure all of our help documents use `User Profiles`, and only use the term `standard_attributes` or `custom_attributes` when we refer to the config / graphQL schema or user import API, and never use the term `attribute`

**Comment by fungc-io** [comment]
Yup, i agree we should use "User Profiles" in the doc menu you shared. It should be "Update User Profile".

It would be less confusing if we describe the structure of a "user profile" with clarity in [https://docs.authgear.com/how-to-guide/user-management/user-profile](https://docs.authgear.com/how-to-guide/user-management/user-profile)

We should in the intro mention explicitly there are 3 components: standard attributes, custom attributes, and roles.

@pius, what do you think?

**Comment by louischan-oursky** [comment]
> We should in the intro mention explicitly there are 3 components: standard attributes, custom attributes, and roles.

Now that we have roles and groups, I think we should say user profile is standard attributes + custom attributes.
